File: org.eclipse.scout.commons/src/main/java/org/eclipse/scout/commons/resource/MimeTypes.java
/*******************************************************************************
 * Copyright (c) 2015 BSI Business Systems Integration AG.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v10.html
 *
 * Contributors:
 *     BSI Business Systems Integration AG - initial API and implementation
 ******************************************************************************/
package org.eclipse.scout.commons.resource;

import java.util.Arrays;
import java.util.Collection;

/**
 * Enumeration for a few well-known mime types.
 */
public enum MimeTypes {
  TEXT_PLAIN("text/plain", "txt"),
  IMAGE_PNG("image/png", "png"),
  IMAGE_JPG("image/jpg", "jpg"),
  IMAGE_JPEG("image/jpeg", "jpg"),
  IMAGE_GIF("image/gif", "gif");

  private final String preferedFileExtension, type;

  MimeTypes(String type, String preferedFileExtension) {
    this.preferedFileExtension = preferedFileExtension;
    this.type = type;
  }

  public String getPreferedFileExtension() {
    return preferedFileExtension;
  }

  public String getType() {
    return type;
  }

  public static MimeTypes convertToMimeType(String mimeType) {
    for (MimeTypes mimeType0 : values()) {
      if (mimeType0.getType().equals(mimeType)) {
        return mimeType0;
      }
    }
    return null;
  }

  /**
   * Common image mime types.
   */
  public static Collection<MimeTypes> getCommonImageTypes() {
    return Arrays.asList(new MimeTypes[]{IMAGE_GIF, IMAGE_JPG, IMAGE_JPEG, IMAGE_PNG});
  }

  public static boolean isOneOf(Collection<MimeTypes> mimeTypes, String mimeType) {
    for (MimeTypes mimeType0 : mimeTypes) {
      if (mimeType0.getType().equals(mimeType)) {
        return true;
      }
    }
    return false;
  }
}

File: org.eclipse.scout.rt.ui.html/src/main/js/scout/util/mimeTypes.js
/*******************************************************************************
 * Copyright (c) 2014-2017 BSI Business Systems Integration AG.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v10.html
 *
 * Contributors:
 *     BSI Business Systems Integration AG - initial API and implementation
 ******************************************************************************/
/**
 * Pendant for org.eclipse.scout.rt.platform.resource.MimeType enumeration.
 */
scout.mimeTypes = {

  TEXT_PLAIN: 'text/plain',
  TEXT_HTML: 'text/html',
  TEXT_CSS: 'text/css',
  TEXT_JAVASCRIPT: 'text/javascript',

  IMAGE_PNG: 'image/png',
  IMAGE_JPG: 'image/jpg',
  IMAGE_JPEG: 'image/jpeg',
  IMAGE_GIF: 'image/gif',

  APPLICATION_JSON: 'application/json',
  APPLICATION_JAVASCRIPT: 'application/javascript',
  APPLICATION_ZIP: 'application/zip',

  JSON: this.APPLICATION_JSON,
  JAVASCRIPT: this.APPLICATION_JAVASCRIPT,
  ZIP: this.APPLICATION_ZIP,

  isTextMimeType: function(mimeType) {
    return scout.isOneOf(mimeType,
      this.TEXT_PLAIN,
      this.TEXT_HTML,
      this.TEXT_CSS,
      this.TEXT_JAVASCRIPT,
      this.APPLICATION_JAVASCRIPT,
      this.APPLICATION_JSON);
  },

  isJavaScript: function(mimeType) {
    return scout.isOneOf(mimeType,
      this.TEXT_JAVASCRIPT,
      this.APPLICATION_JAVASCRIPT);
  }

};
